### PR TITLE
This commit fixes the issue with running subworkflows without defined configfile

### PR DESCRIPTION
**Pull request by SultanOrazbayev**
This commit fixes the issue with running subworkflows without defined configfile configfile, the original fix was suggested by @wclose (see https://github.com/snakemake/snakemake/issues/24). A test for this case has been added to ensure future versions take this into account.

### CODE REVIEW

**Comment by SultanOrazbayev** [comment]
Running of an existing test "test_subworkflows" fails. However, running the latest version of snakemake on that test also fails. Looking at the test, it seems that the root workflow should also contain a `configfile` (or the workflow should be redefined to remove the reference to config dictionary).

**Comment by johanneskoester** [comment]
Actually, that test runs fine on the master branch.

File: snakemake/workflow.py [hunk 1]
```diff
@@ -558,7 +558,7 @@ def files(items):
                         subworkflow.snakefile,
                         workdir=subworkflow.workdir,
                         targets=subworkflow_targets,
-                        configfiles=[subworkflow.configfile],
+                        configfiles=subworkflow.configfile,
```
Comment (johanneskoester) [inline snakemake/workflow.py]: Configfiles has to be a list. 
```suggestion
                        configfiles=[subworkflow.configfile] if subworkflow.configfile else None,
```

**Comment by johanneskoester** [comment]
May I ask you to run black formatting with `black snakemake; black tests/tests.py` on this branch?

**Comment by SultanOrazbayev** [comment]
I ran the commands above, but I'm not sure if the result is correct. After running the commands, the following change was made:
```
-                        configfiles=[subworkflow.configfile] if subworkflow.configfile else None,
+                        configfiles=[subworkflow.configfile]
+                        if subworkflow.configfile
+                        else None,
```
The suggested change doesn't seem to work (or am I missing something?).

**Comment by mwort** [comment]
Any chance this could be merged soon? This is quite an urgent fix for workflows relying on subworkflows.

**Comment by johanneskoester** [comment]
Ok, working on it. The formatting change suggested by black seems fine to me.

**Comment by johanneskoester** [comment]
See PR #271

**Pull request merged**